index.js

```diff
@@ -65,6 +66,10 @@ module.exports = function build (fn, opts = {}) {
 
     // set it to null to not retain a reference to the promise
     res = null
+  } else if (opts.enablePipelining) {
```

Comment:
I would add a check if the user does not return the stream

```suggestion
  } else if (opts.enablePipelining && res) {
```